eZ/Bundle/EzPublishIOBundle/Command/MigrateFilesCommand.php

```diff
@@ -0,0 +1,276 @@
+
+        $this->migrationHandler = $migrationHandler;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:io:migrate-files')
+            ->setDescription('Migrates files from one IO repository to another')
+            ->addOption('from', null, InputOption::VALUE_REQUIRED, 'Migrate from <from_metadata_handler>,<from_binarydata_handler>')
+            ->addOption('to', null, InputOption::VALUE_REQUIRED, 'Migrate to <to_metadata_handler>,<to_binarydata_handler>')
+            ->addOption('list-io-configs', null, InputOption::VALUE_NONE, 'List available IO configurations')
+            ->addOption('remove-files', null, InputOption::VALUE_NONE, 'Remove source files after copying')
+            ->addOption('bulk-count', null, InputOption::VALUE_REQUIRED, 'Number of files processed at once', 100)
+            ->addOption('dry-run', null, InputOption::VALUE_NONE, 'Execute a dry run')
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> migrates files from one IO repository
+to another.
+
+It can for example be used to migrate local files from the default IO
+configuration to a new IO configuration, like a clustered setup.
+
+The <info>--from</info> and <info>--to</info> values must be specified as <info><metadata_handler>,<binarydata_handler></info>.
+If <info>--from</info> is omitted, the default IO configuration will be used.
+If <info>--to</info> is omitted, the first non-default IO configuration will be used.
+
+<fg=red>During the script execution the files should not be modified. To avoid
+surprises you are advised to create a backup and/or execute a dry run before
+proceeding with actual update.</>
+
+Since this script can potentially run for a very long time, to avoid memory
+exhaustion run it in production environment using <info>--env=prod</info> switch.
+
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getOption('list-io-configs')) {
+            $this->outputConfiguredHandlers($output);
+
+            return;
+        }
+
+        $output->writeln($this->getProcessedHelp());
+
+        $fromHandlers = $input->getOption('from') ? explode(',', $input->getOption('from')) : null;
+        $toHandlers = $input->getOption('to') ? explode(',', $input->getOption('to')) : null;
+        if (!$this->areHandlerOptionsValid($fromHandlers, $toHandlers, $output)) {
+            return;
+        }
+
+        if (!$fromHandlers) {
+            $fromHandlers = ['default', 'default'];
+        }
+        if (!$toHandlers) {
+            $toHandlers = [
+                array_keys($this->configuredMetadataHandlers)[0],
+                array_keys($this->configuredBinarydataHandlers)[0],
+            ];
+        }
+
+        $output->writeln([
+            "Migrating from '$fromHandlers[0],$fromHandlers[1]' to '$toHandlers[0],$toHandlers[1]'",
```

Comment:
If you don't wanna bother about quote you can do : 

 sprintf("Migrating from '%s' to '%s'", implode(',', $fromHandlers), implode(',', $toHandlers));

this is a useful trick when arrays are longer.